options/env.go

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0'
+ */
+
+package options
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/canonical/edgex-snap-hooks/v2/files"
+	"github.com/canonical/edgex-snap-hooks/v2/log"
+)
+
+type EnvVarOverrides struct {
+	Service  string
+	Filename string
+	Buffer   *bytes.Buffer
+}
+
+func getEnvVarFile(service string) *EnvVarOverrides {
+	env := EnvVarOverrides{}
+	env.Service = service
+	env.Filename = env.getEnvFilename()
+	env.Buffer = &bytes.Buffer{}
+	return &env
+}
+
+func (env *EnvVarOverrides) setEnvVariable(setting string, value string) error {
+	result := strings.ToUpper(setting)
+	result = strings.Replace(result, "-", "", -1)
+	result = strings.Replace(result, ".", "_", -1)
+	log.Infof("Mapping %s to %s", setting, result)
+	_, err := fmt.Fprintf(env.Buffer, "export %s=%s\n", result, value)
+	return err
+}
+
+func (env *EnvVarOverrides) getEnvFilename() string {
+
+	// Handle security-* service naming. The service names in this
+	// hook historically do not align with the actual binary commands.
+	// As such, when handling configuration settings for them, we need
+	// to translate the hook name to the actual binary name.
+	if env.Service == "security-proxy" {
+		env.Service = "security-proxy-setup"
+	} else if env.Service == "security-secret-store" {
+		env.Service = "security-secretstore-setup"
+	}
```

Comment:
This legacy mapping isn't required because we are adding new options. They are misleading because the options relate to `security-proxy-setup` and `security-secretstore-setup` and not the actual `security-proxy` and `security-secret-store` apps which use different names.

Reply:
removed this mapping